lib/scraped_page_archive.rb

```diff
@@ -14,41 +14,100 @@
 class ScrapedPageArchive
   class Error < StandardError; end
 
-  attr_writer :github_repo_url
+  class GitStorage
```

Comment:
There's a lot going on in here, so I think it's probably worth splitting this out to a separate file.